### PR TITLE
feat: support mmdetection library in Determined

### PR DESCRIPTION
**Pull request by liamcli**
## Description
We now support mmdetection in Determined.  I have verified both speed and accuracy performance within Determined.  See the README for more details.

## Questions
Right now the experiment config references a public docker image that I built on top of our agent image.  A Dockerfile is also provided so users can build their own image if needed.  Is this good enough?

## Checklist
- [x] Add to test_distributed.py
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.

### CODE REVIEW

File: examples/computer_vision/mmdetection/README.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+Most of the configurations provided in the original [`mmdetection` repo](https://github.com/open-mmlab/mmdetection/tree/master/configs) are supported here.  You can see examples of how to submit an experiment to a Determined cluster with a particular config in [`maskrcnn.yaml`](maskrcnn.yaml).  The provided experiment configurations are as follows:
+* Mask R-CNN: [`maskrcnn.yaml`](mask_rcnn.yaml)
+* RetinaNet: [`retinanet.yaml`](retinanet.yaml)
+* Generalized Focal Loss: [`gfl.yaml`](gfl.yaml)
+
+## Performance
+We evaluated the performance of the library in Determined relative to the original mmdetection library both in terms of speed and predictive accuracy.  The comparisons were conducted on `n1-standard-64` instances with 8 Nvidia Tesla V100 GPUs.
+You can get the docker image we used by running
+```
+docker pull liamdetermined/development:mmdetection
```
Comment (katport) [inline examples/computer_vision/mmdetection/README.md:14]: question: what's special about this docker image? Could we just put the libraries in a startup-hook? If not, maybe we should make an official determined docker images for examples.

File: examples/computer_vision/mmdetection/README.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+``` 
+which uses Pytorch 1.4 with CUDA 10.0.  
+
+### Speed Comparison
+We ran the following command within the docker container from the `mmdetection` folder:
+```
+./toos/dist_train.sh configs/mask_rcnn_r50_caffe_fpn_poly_1x_coco.py 8
+```
+both with and without tensorboard logging.  We enabled tensorflow by uncommenting [this line](https://github.com/open-mmlab/mmdetection/blob/master/configs/_base_/default_runtime.py#L7).
```
Comment (armandmcqueen) [inline examples/computer_vision/mmdetection/README.md:23]:
```suggestion
both with and without tensorboard logging.  We enabled tensorboard by uncommenting [this line](https://github.com/open-mmlab/mmdetection/blob/master/configs/_base_/default_runtime.py#L7).  
```

File: examples/computer_vision/mmdetection/README.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+``` 
+which uses Pytorch 1.4 with CUDA 10.0.  
+
+### Speed Comparison
+We ran the following command within the docker container from the `mmdetection` folder:
+```
+./toos/dist_train.sh configs/mask_rcnn_r50_caffe_fpn_poly_1x_coco.py 8
+```
+both with and without tensorboard logging.  We enabled tensorflow by uncommenting [this line](https://github.com/open-mmlab/mmdetection/blob/master/configs/_base_/default_runtime.py#L7).  
+The logs with run times of the first 700 iterations are availble in the [`benchmarks_results`](./benchmark_results) folder.
```
Comment (armandmcqueen) [inline examples/computer_vision/mmdetection/README.md:24]:
```suggestion
The logs with run times of the first 700 iterations are available in the [`benchmarks_results`](./benchmark_results) folder.
```

File: examples/computer_vision/mmdetection/README.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+![Training Metrics](./imgs/train_metrics.png "Training Metrics")
+
+**Validation Metrics**
+
+![Validation Metrics](./imgs/val_metrics.png "Validation Metrics")
+
+
+## Try it out!
+You will need to have access to the COCO 2017 data either via disk or through Google Cloud Storage and set the `backend` filed of the `data` config to either `disk` or `gcp` accordingly.
```
Comment (armandmcqueen) [inline examples/computer_vision/mmdetection/README.md:50]:
```suggestion
You will need to have access to the COCO 2017 data either via disk or through Google Cloud Storage and set the `backend` field of the `data` config to either `disk` or `gcp` accordingly.
```

**Review by naren-determined — COMMENTED** [review]
Can you make sure you have followed these steps for porting an example? https://docs.google.com/document/d/1Ga3SsjZqaRZJslKwuZozNzHS__ovNQqqdxrmqWZLvrM/edit

File: examples/computer_vision/mmdetection/README.md (hunk 1 above)
Comment (liamcli) [reply examples/computer_vision/mmdetection/README.md:14]: Setting up dependencies in the startup-hook takes too long.  I think it makes sense to consider having a docker repo for examples and the model hub.

File: examples/computer_vision/mmdetection/README.md (hunk 2 above)
Comment (liamcli) [reply examples/computer_vision/mmdetection/README.md]: thanks for catching this

File: examples/computer_vision/mmdetection/README.md (hunk 4 above)
Comment (liamcli) [reply examples/computer_vision/mmdetection/README.md]: thanks for catching this

File: examples/computer_vision/mmdetection/README.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+Determined has tensorboard enabled by default so we compare the tensorboard times accordingly:
+|                    | In Determined | Original Repo |
+|--------------------|---------------|---------------|
+| Secs per iteration | 0.310         | 0.305         |
+
+
+### Model Quality
+We trained the [`gfl.yaml`](gfl.yaml) model to convergence and reached a mAP of 39.4. With Determined, you can monitor the performance of your experiments in real time with our WebUI.
+
+**Training Metrics**
```
[the remark this answers is not in the document]
Comment (liamcli) [reply examples/computer_vision/mmdetection/README.md:48]: i grouped both accuracy and speed metrics under results

**Review by naren-determined — COMMENTED** [review]
Can you do steps 5 and 6 of: https://docs.google.com/document/d/1G5pKRllMnNXnLQKYlHGMjOfFz66OHUcGRVnSLVMs8WQ/edit? Otherwise, looks good!

**Review by naren-determined — APPROVED** [review]
🚢
